Modules/binascii.c

```diff
@@ -1639,16 +1649,46 @@ static PyModuleDef_Slot binascii_slots[] = {
     {0, NULL}
 };
 
+static int
+binascii_traverse(PyObject *m, visitproc visit, void *arg)
+{
+    binascii_state *state = get_binascii_state(m);
+    if (state == NULL) {
+        return -1;
+    }
+    Py_VISIT(state->Error);
+    Py_VISIT(state->Incomplete);
+    return 0;
+}
+
+static int
+binascii_clear(PyObject *m)
+{
+    binascii_state *state = get_binascii_state(m);
+    if (state == NULL) {
```

Comment:
I don't think that state can be NULL. If it's NULL, it's a bug in Python. No? Same remark in binascii_traverse().

Reply:
I do some debug in my vm:
```
Breakpoint 1, binascii_traverse (module=0x7ffff7e803d0, visit=0x4630b6 <bad_traverse_test>, arg=0x0) at /temp/shihai/cpython/Modules/binascii.c:1654
1654    {
(gdb) n
1655        binascii_state *state = get_binascii_state(module);
(gdb) n
1656        if (state == NULL) {
(gdb) n
1657            return -1;
(gdb) p state
$5 = (binascii_state *) 0x0
(gdb) c
Continuing.

Breakpoint 3, binascii_exec (module=0x7ffff7e803d0) at /temp/shihai/cpython/Modules/binascii.c:1615
1615    binascii_exec(PyObject *module) {
```

What's the reason? 
the malloc operation of `md_state` will be called after first calling `binascii_traverse`.
https://github.com/python/cpython/blob/master/Objects/moduleobject.c#L399.